src/main/grammars/RustParser.bnf

```diff
@@ -999,11 +999,7 @@ StructLiteral ::= <<checkStructAllowed>> OuterAttr* PathGenericArgsWithColonsNoT
 
 StructLiteralBody ::= '{' StructLiteralField_with_recover* ('..'  AnyExpr)? '}' { pin = 1 }
 
-StructLiteralField ::= OuterAttr* identifier [ ':' AnyExpr ] {
-  pin = 2
-  implements = [ "org.rust.lang.core.psi.ext.RsReferenceElement" ]
-  mixin = "org.rust.lang.core.psi.ext.RsStructLiteralFieldImplMixin"
-}
+StructLiteralField ::= OuterAttr* ((FieldName | INTEGER_LITERAL) ':' AnyExpr | PathExpr)
```

Comment:
I guess we can implement it without making PSI more complicated. I.e. write here `identifier | INTEGER_LITERAL` and make `INTEGER_LITERAL` `referenceNameElement` as well.